src/rust/engine/fs/store/src/lib.rs

```diff
@@ -69,17 +69,48 @@ pub struct UploadSummary {
   pub upload_wall_time: Duration,
 }
 
-#[derive(Clone, Debug, PartialEq, Eq, Serialize)]
-pub enum Source {
-  Local,
-  Remote,
+/// A timespan represented as two floating point representations of seconds
+#[derive(Copy, Clone, Debug, PartialEq, Serialize)]
+pub struct TimeSpan {
```

Comment:
This type currently represents both an `Instant` (when it's only been started) and a pair of `Instant` and `Duration` (when it's also been ended), which is a little weird to reason about. Rather than having `start` return a partial `TimeSpan`, maybe we could phrase it something like:
```rust
let start = std::time::SystemTime::now();
// Do some stuff
let timespan = TimeSpan::since(start);
```

WDYT?

Reply:
Yes, that makes sense.